src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.add-email-address.unit.spec.jsx

```diff
@@ -0,0 +1,264 @@
+import React from 'react';
+import { MemoryRouter } from 'react-router-dom';
+import { waitForElementToBeRemoved } from '@testing-library/react';
+import user from '@testing-library/user-event';
+import { expect } from 'chai';
+import { setupServer } from 'msw/node';
+
+import { resetFetch } from 'platform/testing/unit/helpers';
+
+import * as mocks from '../../../msw-mocks';
+import {
+  getBasicContactInfoState,
+  renderWithProfileReducers,
+} from '../../unit-test-helpers';
+
+import PersonalInformation from '../../../components/personal-information/PersonalInformation';
+
+function createBasicInitialState() {
+  return {
+    scheduledDowntime: {
+      globalDowntime: null,
+      isReady: true,
+      isPending: false,
+      serviceMap: {
+        get() {},
+      },
+      dismissedDowntimeWarnings: [],
+    },
+    vaProfile: {
+      personalInformation: {
+        gender: 'M',
+        birthDate: '1986-05-06',
+      },
+    },
+    user: {
+      profile: {
+        vet360: getBasicContactInfoState(),
+      },
+    },
+  };
+}
+
+describe('Adding email address', () => {
+  // before we can use msw, we need to make sure that global.fetch has been
+  // restored and is no longer a sinon stub.
+  let server;
+  before(() => {
+    resetFetch();
+    server = setupServer(...mocks.addEmailAddressSuccess());
+    server.listen();
+  });
+  beforeEach(() => {
+    window.VetsGov = { pollTimeout: 1 };
```

Comment:
What does this `pollTimeout` do?

Reply:
This is a convention that's been used in the past to determine what to set timeout/interval delays as so we can use a much shorter delay when running unit tests. In this particular case this shortens the polling delay when we check on the status of a pending transaction. Under normal situations we check on the status of the transaction every 1000ms, but in unit tests we'll check every 1ms. I'm also using the existence of this global flag to determine if our redux actions creators should make real API calls (in `isVet360Configured()`) and changing how aggressively we automatically exit edit mode when a save is taking a long time (ie, we change the definition of what a "long time" means in the context of waiting for a profile save to happen from 5000ms to 50ms).